pandas/core/arrays/base.py

```diff
@@ -248,6 +248,41 @@ def unique(self):
         uniques = unique(self.astype(object))
         return self._constructor_from_sequence(uniques)
 
+    def factorize(self, na_sentinel=-1):
+        """Encode the extension array as an enumerated type.
+
+        Parameters
+        ----------
+        na_sentinel : int, default -1
+            Value to use in the `labels` array to indicate missing values.
+
+        Returns
+        -------
+        labels : ndarray
+            An interger NumPy array that's an indexer into the original
+            ExtensionArray
+        uniques : ExtensionArray
+            An ExtensionArray containing the unique values of `self`.
+
+        See Also
+        --------
+        pandas.factorize : top-level factorize method that dispatches here.
+
+        Notes
+        -----
+        :meth:`pandas.factorize` offers a `sort` keyword as well.
+        """
+        from pandas.core.algorithms import _factorize_array
```

Comment:
We're OK with using this private API here? 

Because an extension authors might want to copy paste this method and change the `arr = self.astype(object)` line? (how do you implement `factorize` in cyberpandas?)

Reply:
Quite similar to this. https://github.com/ContinuumIO/cyberpandas/blob/468644bcbdc9320a1a33b0df393d4fa4bef57dd7/cyberpandas/base.py#L72

In that case I think going to object dtypes is unavoidable, since there's no easy way to factorize a 2-D array, and I didn't want to write a new hashtable implementation :)

Reply:
w.r.t. using factorize array, I don't think it's avoidable.

We might consider making it public / semi-public (keep the `_` to not confuse users, but point EA authors to it).